src/components/Select/components/SelectControl/SelectControl.tsx

```diff
@@ -45,12 +55,66 @@ export const SelectControl = React.forwardRef<HTMLElement, ControlProps>((props,
         open,
         disabled,
         value,
+        hasClear,
     } = props;
     const controlRef = React.useRef<HTMLElement>(null);
     const handleControlRef = useForkRef<HTMLElement>(ref, controlRef);
     const showOptionsText = Boolean(selectedOptionsContent);
     const showPlaceholder = Boolean(placeholder && !showOptionsText);
-    const mods: CnMods = {open, size, view, pin, disabled, error: Boolean(error)};
+    const hasValue = Array.isArray(value) && !isEmpty(value.filter(Boolean));
+
+    const [isDisabledButtonAnimation, setIsDisabledButtonAnimation] = React.useState(false);
+
+    const controlMods: CnMods = {
+        open,
+        size,
+        pin,
+        disabled,
+        error: Boolean(error),
+        'has-clear': hasClear,
+        'no-active': isDisabledButtonAnimation,
+        'has-value': hasValue,
+    };
+
+    const buttonMods: CnMods = {
+        open,
+        size,
+        view,
+        pin,
+        disabled,
+        error: Boolean(error),
+    };
+
+    const disableButtonAnimation = React.useCallback(() => {
+        setIsDisabledButtonAnimation(true);
+    }, []);
+    const enableButtonAnimation = React.useCallback(() => {
+        setIsDisabledButtonAnimation(false);
+    }, []);
+    const handleOnClearIconClick = React.useCallback(
+        (e: React.MouseEvent) => {
+            // return animation on clear click
+            setIsDisabledButtonAnimation(false);
+            clearValue?.(e);
```

Comment:
Why clearValue is optional? It is our inner method and we always add them to the component